pkg/operator/configobserver/network/observe_network.go

```diff
@@ -3,110 +3,57 @@ package network
-	"k8s.io/client-go/listers/core/v1"
 
 	"github.com/openshift/library-go/pkg/operator/events"
 )
 
-const (
-	clusterConfigNamespace = "kube-system"
-	clusterConfigName      = "cluster-config-v1"
-)
-
-// GetClusterCIDRs reads the cluster CIDRs from the install-config ConfigMap in the cluster.
-func GetClusterCIDRs(lister v1.ConfigMapLister, recorder events.Recorder) ([]string, error) {
-	clusterConfig, err := lister.ConfigMaps(clusterConfigNamespace).Get(clusterConfigName)
+// GetClusterCIDRs reads the cluster CIDRs from the global network configuration resource. Emits events if CIDRs are not found.
+func GetClusterCIDRs(lister configlistersv1.NetworkLister, recorder events.Recorder) ([]string, error) {
+	network, err := lister.Get("cluster")
 	if errors.IsNotFound(err) {
-		recorder.Warningf("ObserveClusterCIDRFailed", "Required %s/%s config map not found", clusterConfigNamespace, clusterConfigName)
-		glog.Warning("configmap/cluster-config-v1.kube-system: not found")
+		recorder.Warningf("ObserveRestrictedCIDRFailed", "Required networks.%s/cluster not found", configv1.GroupName)
 		return nil, nil
 	}
 	if err != nil {
+		recorder.Warningf("ObserveRestrictedCIDRFailed", "error getting networks.%s/cluster: %v", configv1.GroupName, err)
 		return nil, err
 	}
 
-	installConfigYaml, ok := clusterConfig.Data["install-config"]
-	if !ok {
-		glog.Warning("configmap/cluster-config-v1.kube-system: install-config not found")
-		recorder.Warningf("ObserveClusterCIDRFailed", "ConfigMap %s/%s does not have required 'install-config'", clusterConfigNamespace, clusterConfigName)
-		return nil, nil
-	}
-	installConfig := map[string]interface{}{}
-	err = yaml.Unmarshal([]byte(installConfigYaml), &installConfig)
-	if err != nil {
-		recorder.Warningf("ObserveRestrictedCIDRFailed", "Unable to decode install config: %v'", err)
-		return nil, fmt.Errorf("unable to parse install-config: %s", err)
+	if len(network.Spec.ClusterNetwork) == 0 {
+		recorder.Warningf("ObserveClusterCIDRFailed", "Required spec.clusterNetwork field is not set in networks.%s/cluster", configv1.GroupName)
+		return nil, fmt.Errorf("networks.%s/cluster: spec.clusterNetwork not found", configv1.GroupName)
 	}
 
 	var clusterCIDRs []string
-	clusterNetworks, _, err := unstructured.NestedSlice(installConfig, "networking", "clusterNetworks")
-	if err != nil {
-		return nil, fmt.Errorf("unabled to parse install-config: %s", err)
-	}
-	for i, n := range clusterNetworks {
-		obj, ok := n.(map[string]interface{})
-		if !ok {
-			recorder.Warningf("ObserveRestrictedCIDRFailed", "Required networking.clusterNetworks field is not set in install-config")
-			return nil, fmt.Errorf("unabled to parse install-config: expected networking.clusterNetworks[%d] to be an object, got: %#v", i, n)
-		}
-		cidr, _, err := unstructured.NestedString(obj, "cidr")
-		if err != nil {
-			return nil, fmt.Errorf("unabled to parse install-config: %v", err)
+	for i, clusterNetwork := range network.Spec.ClusterNetwork {
```

Comment:
in  the cold light of morning.  How about the status one per the doc https://github.com/openshift/api/blob/master/config/v1/types_network.go#L22